src/core/meshcop/dtls.cpp

```diff
@@ -317,23 +361,7 @@ void Dtls::Close(void)
     mTimer.Start(kGuardTimeNewConnectionMilli);
     VerifyOrExit(mStarted);
```

Comment:
If Dtls::Start fails before mStarted = true , please check you will release the resource or not.